ngraph/core/reference/src/runtime/reference/experimental_detectron_detection_output.cpp

```diff
@@ -0,0 +1,421 @@
+//*****************************************************************************
+// Copyright 2017-2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/experimental_detectron_detection_output.hpp"
+#include <algorithm>
+#include <cassert>
+#include <utility>
+#include "ngraph/runtime/reference/experimental_detectron_detection_output.hpp"
+#include "ngraph/shape.hpp"
+
+namespace
+{
+    void refine_boxes(const float* boxes,
+                      const float* deltas,
+                      const float* weights,
+                      const float* scores,
+                      float* refined_boxes,
+                      float* refined_boxes_areas,
+                      float* refined_scores,
+                      const int64_t rois_num,
+                      const int64_t classes_num,
+                      const float img_H,
+                      const float img_W,
+                      const float max_delta_log_wh,
+                      float coordinates_offset)
+    {
+        for (int64_t roi_idx = 0; roi_idx < rois_num; ++roi_idx)
```

Comment:
Am I right that it is a copy past from CPU plugin? or do we have any differences?

Reply:
Yes, we have differences, but these differences are small: `int64_t` in `ngraph\core\reference\src\runtime\reference\experimental_detectron_detection_output.cpp` instead of `int` in CPU implementation, and `ngraph\core\reference\src\runtime\reference\experimental_detectron_detection_output.cpp` uses explicit index calculations instead of the class `Indexer` in the CPU implementation, because these index calculations are simple.